src/zkp/piaffg.rs

```diff
@@ -28,103 +43,146 @@ use tracing::warn;
+    /// the paper).
+    random_add_coeff_commit: Commitment,
+    /// The Fiat-Shamir challenge value (`e` in the paper).
+    challenge: BigNumber,
+    /// A mask of the (secret) multiplicative coefficient (`z_1` in the paper).
+    masked_mult_coeff: BigNumber,
+    /// A mask of the (secret) additive coefficient (`z_2` in the paper).
+    masked_add_coeff: BigNumber,
+    /// A mask of the commitment randomness of the (secret) multiplicative
+    /// coefficient (`z_3` in the paper).
+    masked_mult_coeff_commit_randomness: MaskedRandomness,
+    /// A mask of the commitment randomness of the (secret) additive coefficient
+    /// (`z_4` in the paper).
+    masked_add_coeff_commit_randomness: MaskedRandomness,
+    /// A mask of the Paillier ciphertext nonce of the (secret) additive
+    /// coefficient under the 0th encryption key (`w` in the paper).
+    masked_add_coeff_nonce_0: MaskedNonce,
+    /// A mask of the Paillier ciphertext nonce of the (secret) additive
+    /// coefficient under the 1st encryption key (`w_y` in the paper).
+    masked_add_coeff_nonce_1: MaskedNonce,
 }
 
+/// Common input and setup parameters for [`PiAffgProof`] known to both the
+/// prover and verifier.
 #[derive(Serialize)]
 pub(crate) struct PiAffgInput {
-    setup_params: VerifiedRingPedersen,
-    g: CurvePoint,
-    /// This corresponds to `N_0` in the paper.
-    pk0: EncryptionKey,
-    /// This corresponds to `N_1` in the paper.
-    pk1: EncryptionKey,
-    C: Ciphertext,
-    D: Ciphertext,
-    Y: Ciphertext,
-    X: CurvePoint,
+    /// The verifier's commitment parameters (`(Nhat, s, t)` in the paper).
+    verifier_setup_params: VerifiedRingPedersen,
+    /// First Paillier encryption key (`N_0` in the paper).
+    encryption_key_0: EncryptionKey,
+    /// Second Paillier encryption key (`N_1` in the paper).
+    encryption_key_1: EncryptionKey,
+    /// Input Paillier ciphertext (`C` in the paper).
+    input_ciphertext: Ciphertext,
+    /// Output Paillier ciphertext (`D` in the paper).
+    output_ciphertext: Ciphertext,
+    /// Paillier ciphertext of the prover's additive coefficient under the 1st
+    /// encryption key (`Y` in the paper).
+    add_coeff_ciphertext_1: Ciphertext,
+    /// Exponentiation of the prover's multiplicative coefficient (`X` in the
+    /// paper).
+    mult_coeff_exp: CurvePoint,
 }
 
 impl PiAffgInput {
+    /// Construct a new [`PiAffgInput`] type.
     #[allow(clippy::too_many_arguments)]
     pub(crate) fn new(
-        setup_params: &VerifiedRingPedersen,
-        g: &CurvePoint,
-        pk0: &EncryptionKey,
-        pk1: &EncryptionKey,
-        C: &Ciphertext,
-        D: &Ciphertext,
-        Y: &Ciphertext,
-        X: &CurvePoint,
+        verifier_setup_params: VerifiedRingPedersen,
+        encryption_key_0: EncryptionKey,
+        encryption_key_1: EncryptionKey,
+        input_ciphertext: Ciphertext,
+        output_ciphertext: Ciphertext,
+        add_coeff_ciphertext_1: Ciphertext,
```

Comment:
suggestion: These are inconsistent about denoting which key they're encrypted under. I'm not sure what the best approach is for naming, though.